### ISSUE TITLE
Elements pools

### ISSUE THREAD

**Issue by CrosRoad95** [multitheftauto/mtasa-blue#1880]

**Comment by Pirulax** [comment]
I think we should work on the root cause of the issue instead of trying to do workarounds.
I this case the root cause is probably the packet sending, although I should profile it, to make sure.

Note: Manually moving a vehicle out of the dimension (and then back) back does the same thing, so having element pools in Lua is totally possible.

**Comment by PlatinMTA** [comment]
> I think we should work on the root cause of the issue instead of trying to do workarounds.

I agree with you, apart that you can also do this in lua and its kind of a hack.

> I this case the root cause is probably the packet sending, although I should profile it, to make sure.

It does take a considerable amount of time even on empty servers, but that only seems to happen serverside.

**Comment by CrosRoad95** [comment]
both - improve create/destroy performance, and add elements pools,
even unity3d even if is well optimized, people implements objects pool

**Comment by Pirulax** [comment]
> It does take a considerable amount of time even on empty servers, but that only seems to happen serverside.

Yes, because the packet is constructed even if there are no players on the server, just that it's not sent. This could only be solved by moving all the packet stuff to a separate thread.

> I agree with you, apart that you can also do this in lua and its kind of a hack.

Well, it would be a hack in C++ as well, so :D

**Comment by PlatinMTA** [comment]
> Well, it would be a hack in C++ as well, so :D

yeah, thats what i tried to say xD

**Comment by MegadreamsBE** [comment]
> Note: Manually moving a vehicle out of the dimension (and then back) back does the same thing, so having element pools in Lua is totally possible.  

It's very prone to glitching over time for both vehicles and objects due to various of smaller issues within MTA.
Source: Had a element pool but disabled it recently due to increased bugs.

**Comment by Pirulax** [comment]
The problem is that element pools can't be implemented any better than the above mentioned solution.
